librz/include/rz_util/rz_print.h

```diff
@@ -41,6 +42,62 @@ typedef const char *(*RzPrintSectionGet)(void *user, ut64 addr);
 typedef const char *(*RzPrintColorFor)(void *user, ut64 addr, bool verbose);
 typedef char *(*RzPrintHasRefs)(void *user, ut64 addr, int mode);
 
+typedef enum {
+	RZ_ASM_TOKEN_UNKNOWN = 0, //< Does not fit to any token below.
+	RZ_ASM_TOKEN_MNEMONIC, //< Asm mnemonics like: mov, push, lea...
+	RZ_ASM_TOKEN_OPERATOR, //< Arithmetic operators: +,-,<< etc.
+	RZ_ASM_TOKEN_NUMBER, //< Numbers
+	RZ_ASM_TOKEN_REGISTER, //< Registers
+	RZ_ASM_TOKEN_SEPARATOR, //< Brackets, comma etc.
+	RZ_ASM_TOKEN_META, //< Meta information (e.g Hexagon packet prefix, ARM & Hexagon number prefix).
+
+	RZ_ASM_TOKEN_LAST,
+} RzAsmTokenType;
+
+/**
+ *  \brief A token of an asm string holding meta data.
+ */
+typedef struct {
+	size_t start; //< byte-offset into `str` where this token starts. Must be exactly at a utf-8 codepoint boundary.
+	size_t len; //< `str` length of token in bytes.
+	RzAsmTokenType type;
+	union {
+		ut64 number; //< Number of RZ_ASM_TOKEN_NUMBER
+	} val;
+} RzAsmToken;
+
+/**
+ * \brief An tokenized asm string.
+ */
+typedef struct {
+	ut32 op_type; ///< RzAnalysisOpType. Mnemonic color depends on this.
+	RzStrBuf *str; //< Contains the raw asm string
+	RzVector /* <RzAsmToken> */ *tokens; //< Contains only the tokenization meta-info without strings, ordered by start for log2(n) access
+} RzAsmTokenString;
+
+typedef struct {
+	const RzRegSet *reg_sets; ///< Array of reg sets used to lookup register names during parsing.
+	ut32 ana_op_type; ///< Analysis op type (see: _RzAnalysisOpType) of the token string to parse.
+} RzAsmParseParam;
+
+/**
+ * \brief Pattern for a asm string token.
+ */
+typedef struct {
+	RzAsmTokenType type; //< Asm token type.
+	char *pattern; //< The regex pattern describing the tokens.
+	RzRegex *regex; //< Compiled regex pattern.
+} RzAsmTokenPattern;
```

Comment:
should these structs really be in RzPrint? They are also called **RzAsm**something so maybe they belong to RzAsm?

WDYT?

Reply:
I agree that they should go to RzAsm modukle. BUt `rz_print_colorize_asm_str` needs the RzAsmTokenStr as type.
So I see
- Give a `const void *` to  `rz_print_colorize_asm_str` instead of `RzAsmTokenString*` (bit ugly)
- or those structs blong to `RzPrint` (which doesn't fit that well).
Or is there a third way?

Reply:
move `rz_print_colorize_asm_str` to RzCore?